diff --git a/libil2cpp/vm/Runtime.cpp b/libil2cpp/vm/Runtime.cpp
index 276d92d..a6e1b81 100644
--- a/libil2cpp/vm/Runtime.cpp
+++ b/libil2cpp/vm/Runtime.cpp
@@ -62,6 +62,8 @@
 #include "Baselib.h"
 #include "Cpp/ReentrantLock.h"
 
+#include "zlua/lvm/LuaAppDomain.h"
+
 #if IL2CPP_MONO_DEBUGGER
 extern "C" {
 #include <mono/metadata/profiler-private.h>
@@ -401,6 +403,7 @@ namespace vm
         vm::MetadataCache::ExecuteEagerStaticClassConstructors();
         vm::MetadataCache::ExecuteModuleInitializers();
 
+        zlua::LuaAppDomain::Initialize();
         return true;
     }
 
